### PR TITLE
add book.rb && add schema.sql

### LINKED ISSUE #41: [1pt]Create a Book class in a separate .rb file.
Create a book.rb file and initialize a class for it .

### LINKED ISSUE #44: [1pt] Create a Label class with an association to the Item class (in a separate .rb file).
create a label.rb nested with item.rb. Initialize it as a class.

### LINKED ISSUE #45: [1pts] All Label class properties visible in the diagram should be defined and set up in the constructor method.
# Label class constructor method

### LINKED ISSUE #46: [1 pt] add_item method in the Label class should take an instance of the Item class as an input
add add_items to the @items array

### LINKED ISSUE #47: [1pt] add_item method in the Label class: should add the input item to the collection of items
add_item method in the Label class: should add the input item to the collection of items
and set the item's label property to self

### LINKED ISSUE #48: [1pt] add_item method in the Label class: should add self as a property of the item object (by using the correct setter from the item object)
### add_item the method in the Label class : 

define add_item with item property and add into array item with label compared with self.

### LINKED ISSUE #51: [1pt ] can_be_archived?() in the Book class -----should override the method from the parent class: should return true if parent's method returns true OR if cover_state equals to "bad"
need to pass the blocks to the superclass method outside your create() method

### LINKED ISSUE #52: [1 pt ] can_be_archived?() in the Book class : otherwise should return false
define can_be_archived?() if else as false.

### LINKED ISSUE #53: [1pts] Add unit tests for all implemented methods.
Requiring all files arguments to apply test into every class (Book, Music, Movies)  assert if every value with 1 and his argument are false or true. To prove every unit.

### LINKED ISSUE #59: [1pts]Create a schema.sql file with tables that will be analogical to the structure of the classes that you created: books table (add all properties and associations from the parent Item class as table columns)
Provide tables that will be analogical to the structure of the classes that you created: 
 - books table (add all properties and associations from the parent Item class as table columns)

### PR DESCRIPTION
**Pull request by andym80**
- close #41
- close #44
- close #45
- close #46
- close #48
- close #51
- close #52
- close #53
- close #47 
- close #59

### CODE REVIEW

File: src/book.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  def initialize(genre, author, source, label, publish_date, publisher, cover_state) # rubocop:disable Metrics/ParameterLists
+    super(genre, author, source, label, publish_date)
+    @publisher = publisher
+    @cover_state = cover_state
+  end
+
+  def self.list_books(books)
+    books.each_with_index do |book, index|
+      if book_instance_of? Book
+        puts "[#{index}] The Book: #{book.cover_state} by #{book.author.first_name}
```
Comment (AmaliaGM) [inline src/book.rb:19]: Please fix book.cover_state, update it to book.label.

File: src/book.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+         #{book.author.last_name} has been Published by #{book.publisher} on #{book.publish_date}"
+      end
+    end
+  end
+
+  def self.add_book
```
Comment (AmaliaGM) [inline src/book.rb:25]: Please add book.label

File: src/book.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    new_author.add_item(new_book)
+
+    new_book
+  end
+
+  def to_json(*args)
+    {
+      JSON.create_id => self.class.name,
+      'publisher' => @publisher,
+      'cover_state' => @cover_state
```
Comment (AmaliaGM) [inline src/book.rb:51]: Please update to book.label

File: src/label.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  def initialize(title, color, id = SecureRandom.random_number(1_000))
+    @title = title
+    @color = color
+    @id = id
+    @items = []
+  end
+
+  def add_item(item)
+    @items << item
+  end
```
Comment (carlosmuerto) [inline src/label.rb:16]: when adding a label, should update the label property in the item

**Review by AmaliaGM — CHANGES_REQUESTED** [review]

File: src/label.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+  def self.list_labels(things)
+    labels = []
+    things.each do |thing|
+      labels << thing.label
+      labels.each_with_index do |label, index|
+        puts "[#{index + 1}]. (ID: #{label.id}) #{label.title})"
+      end
```
Comment (carlosmuerto) [inline src/label.rb:24]: same as above return a string rather than printing in the console

**Review by carlosmuerto — CHANGES_REQUESTED** [review]
- please add a description and add the issue this change gonna close with the following format
```
- close #<number of the issue>
```

- Add some Units test to ensure that all is working properly before emerging to dev.

**Review by AmaliaGM — APPROVED** [review]

File: src/label.rb (hunk 4 above)
Comment (andym80) [reply src/label.rb:16]: @carlosmuerto check it !

**Review by carlosmuerto — APPROVED** [review]

**Pull request merged**